spanner/spanner_commit_timestamp/main.go

```diff
@@ -0,0 +1,356 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Package main contains runnable code demonstrating Cloud Spanner's commit timestamp feature.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"golang.org/x/net/context"
+	"google.golang.org/api/iterator"
+
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+type command func(ctx context.Context, w io.Writer, client *spanner.Client) error
+type adminCommand func(ctx context.Context, w io.Writer, adminClient *database.DatabaseAdminClient, database string) error
+
+var (
+	commands = map[string]command{
+		"write":             write,
+		"update":            update,
+		"query":             query,
+		"writewithhistory":  writeWithHistory,
+		"updatewithhistory": updateWithHistory,
+		"querywithhistory":  queryWithHistory,
+	}
+
+	adminCommands = map[string]adminCommand{
+		"createdatabase":              createDatabase,
+		"createtablewithtimestamp":    createTableWithTimestamp,
+		"createtablewithhistorytable": createTableWithHistoryTable,
+	}
+)
+
+func createDatabase(ctx context.Context, w io.Writer, adminClient *database.DatabaseAdminClient, db string) error {
+	matches := regexp.MustCompile("^(.*)/databases/(.*)$").FindStringSubmatch(db)
```

Comment:
Ugly to have regexp in a code sample. Can these just be parameters in the function instead?

Can move the regexp into main() then.